src/components/FormListBox/index.tsx

```diff
@@ -0,0 +1,159 @@
+import { ReactNode, useState, useEffect, forwardRef } from "react";
+import { Listbox } from "@headlessui/react";
+import ChevronDownIcon from "../../../public/images/icons/16px/chevronDown.svg";
+import CheckmarkIcon from "../../../public/images/icons/16px/checkmark.svg";
+
+export interface SelectOptionType {
+  name: ReactNode;
+  value: string | number;
+  disabled?: boolean;
+}
+
+interface FormListBoxPropType {
+  name: string;
+  label?: ReactNode;
+  placeholder?: string;
+  className?: string;
+  options: SelectOptionType[];
+  errors?: string[];
+  value?: string | number;
+  defaultValue?: string | number;
+  onChange?: (name: string | number | undefined) => void;
+  containsIconList?: boolean;
+}
+
+// eslint-disable-next-line react/display-name
+export const FormListBox = forwardRef<HTMLButtonElement, FormListBoxPropType>(
+  (
+    {
+      label,
+      options,
+      placeholder = "Bitte wähle eine Option",
+      className = "",
+      errors = [],
+      value,
+      defaultValue,
+      onChange = () => undefined,
+      containsIconList = false,
+    },
+    ref
+  ) => {
+    const [selectValue, setSelectValue] = useState<string | number | undefined>(
+      value || defaultValue
+    );
+
+    useEffect(() => {
+      value && setSelectValue(value);
+    }, [value]);
+
+    const selectValueObject = options.find(
+      ({ value: val }) => val === selectValue
+    );
+
+    return (
+      <div className={`${className} mb-2 relative`}>
+        <Listbox
+          value={options.find(({ value }) => value === selectValue)}
+          onChange={newVal => {
+            if (typeof onChange === "function") onChange(newVal?.value);
+            setSelectValue(newVal?.value);
+          }}
+        >
+          {label && (
+            <Listbox.Label className='block mb-2 cursor-pointer transition hover:opacity-60'>
```

Comment:
So this is how we use `headless-ui`? It gives us a completely unstyled element and we use Tailwind to style it?

Reply:
Exactly, and it takes care of all the aria labels, keyboard navigation, etc. Very cool!